sig-scalability/2020-annual-report.md

```diff
@@ -0,0 +1,112 @@
+
+**Is there a healthy onboarding and growth path for contributors in your SIG?
+What are some activities that the group does to encourage this?
+What programs are you participating in to grow contributors throughout the contributor ladder?**
+
+We maintain a list of `help-wanted` issues and recommend particular ones for people
+interested in conctributing.
+
+**What programs do you participate in for new contributors?**
+
+None
+
+**Does the group have contributors from multiple companies/affiliations?
+Can end users/companies contribute in some way that they currently are not?**
+
+Yes - there are people from multiple companies.
+End users mostly come with questions, they rarely want to contribute.
+
+
+## Current initiatives and project health
+
+**What are initiatives that should be highlighted, lauded, shout out, that your group is proud of?
+Currently underway? What are some of the longer tail projects that your group is working on?**
+
+The critical initiative in 2020 was speeding up scalability tests. We managed to reduce time
+take by our 5k-node scalability tests from ~14h to less than 5h (~3x).
+
+We helped validating scalability and performance impact of multiple features across the whole year
+(by consulting multiple contributors from across many different SIGs).
+
+We also visibly improved the testing framework to support other environments, tracking new
+scalability/performance aspects (e.g. as OOM-killing) and extended portfolio of tested features.
+
+Given SIG-Scalability doesn't own non-test-related code, many efforts are driven under
+the auspicies of other SIGs. This includes:
+- [Efficient Watch Resumption] with sig-api-machinery (Alpha 1.20)
+- [Immutable Secrets and ConfigMaps] with sig-storage (Alpha 1.18, Beta 1.19)
+
+[Efficient Watch Resumption]: https://github.com/kubernetes/enhancements/issues/1904
+[Immutable Secrets and ConfigMaps]: https://github.com/kubernetes/enhancements/issues/1412
+
+
+**Year to date KEP work review: What’s now stable? Beta? Alpha? Road to alpha?**
+
+Given that SIG-Scalability doesn't really own code (other than test frameworks and tests
+themselves), there aren't SIG-Scalability KEPs per-se. However, we initiate and drive many
+KEPs that are officially owned by other SIGs though (and SIG-Scalability is listed as
+participating SIG even though it's often doing most of the work - examples above).
+
+**What areas and/or subprojects does the group need the most help with?**
+
+Each subproject could benefit from additional hands.
+However, the `Scalability Test Frameworks` and `Scalability and Performance tests and validation`
+are the ones where we can grow contributors - the other require both very deep and wide
+understanding of Kubernetes before making reasonable contributions.
+
+**What's the average open days of a PR and Issue in your group? / what metrics does your group care about and/or measure?**
+
+TODO: How to easily answer it post-factum?
```

Comment:
[Devstats](https://k8s.devstats.cncf.io/d/12/dashboards?orgId=1&refresh=15m) is where we pull a lot of metrics, but there are some current limitations that make it difficult to tease out information for SIGs that work out of k/k or other large shared repos =/

This is something ContribEx is working on with the CNCF.

Some of the devstats [PR](https://k8s.devstats.cncf.io/d/12/dashboards?viewPanel=9&orgId=1&refresh=15m) and [Issue](https://k8s.devstats.cncf.io/d/12/dashboards?viewPanel=8&orgId=1&refresh=15m) velocity boards might be useful from a general health standpoint, but if they're not - don't feel the need to include them.

If you have some metrics you'd like to see, something that could potentially help you manage your workload or SIG better, that would also be a great answer for this one.